elastic/kibana
# [SecuritySolution] Add Service entity type to Entity Analytics

## Summary
This ticket is part of the Service Entity Store Epic. Tickets add the service entity to the entity store enablement flow.


* Refactor types to prevent usages of `host` and `user`. 
  * Use `EntityType` instead. 
  * Use a generic function that receives `EntityType` as a parameter instead of custom user and host functions
* Consolidate duplicated entity types
* Add service to entity types and update all references on the EntityAnalyticsDashboards page, Risk score page and Entity Store page.
* Refactor Risk score APIs to be more generic and accept EntityType and a param
* Refactor if statement like `isUserRiskScore` to be more generic and accept `service`
* Delete `RiskScoreEntity` in favour of `EntityType`.
* Update the branch to support the universal entity

### Not included
* Service Flyout

### Images

![Screenshot 2025-01-06 at 15 57 18](https://github.com/user-attachments/assets/a79444c0-d0a8-4838-bea0-5f0afdb58df4)
![Screenshot 2025-01-06 at 16 02 03](https://github.com/user-attachments/assets/1d007591-7ba5-416a-96b3-dc72db63d87b)
![Screenshot 2025-01-06 at 16 08 22](https://github.com/user-attachments/assets/014c49c0-ea6c-4e9a-bb88-75c862c16dd8)

### Generic Entity Support
We need to support risk score and asset criticality for Generic/Universal entities according to https://github.com/elastic/security-team/issues/10740

> We expect that the below will be supported:
> 
> Entity flyout for service/generic entity
> Entity risk scoring for service/generic entity
> Asset criticality assignments for service/generic entity

This PR already implements that support. However, I have introduced a function per feature that returns the enabled entity types. At the moment, I defined universal/generic entities as unsupported for this PR to preserve the current behaviour. But to allow universal/generic entities, we only need to delete a couple of lines.

Risk Score will need extra work because the entity types are hard-coded on some parts of the code.

### How to test it
1
* Start kabana with security solution data 
  * You can use the document generator with `yarn start entity-store`
* Enable Entity Store and Risk engine
* Test the EA features, and they should work normally


2
* Start kabana with security solution data 
  * You can use the document generator with `yarn start entity-store`
* Enable the `serviceEntityStoreEnabled`  flag
* Enable Entity Store and Risk engine
* Test the EA features, and you should see a new type of Entity called 'service'
* Service Entity should work with all Entity analytics features


3 
* Start kabana with security solution data 
  * You can use the document generator with `yarn start entity-store`
* Enable the `assetInventoryStoreEnabled` flag
* Enable Entity Store and Risk engine
* Test the EA features, and you should not see universal/generic entity except for the entity store status pages




### Checklist

Reviewers should verify this PR satisfies this list as well.


## Release note
Add the "service" type to  Security Entity Analytics - Entity Store. It will find services by the `service.name` field, calculate risk score, and allow asset criticality assignment.
